contracts/FlowIDTableStaking.cdc

```diff
@@ -927,15 +911,35 @@ pub contract FlowIDTableStaking {
     }
 
     /// borrow a reference to to one of the nodes in the record
-    access(contract) fun borrowNodeRecord(_ nodeID: String): &NodeRecord {
+    access(account) fun borrowNodeRecord(_ nodeID: String): &NodeRecord {
         pre {
             FlowIDTableStaking.nodes[nodeID] != nil:
                 "Specified node ID does not exist in the record"
         }
         return &FlowIDTableStaking.nodes[nodeID] as! &NodeRecord
     }
 
-    /****************** Getter Functions for the staking Info *******************/
+    access(account) fun updateClaimed(path: StoragePath, _ key: String, claimed: Bool) {
+        let claimedDictionary = self.account.load<{String: Bool}>(from: path)
+            ?? panic("Invalid path for dictionary")
+
+        if claimed {
+            claimedDictionary[key] = true
+        } else {
+            claimedDictionary[key] = nil
+        }
+
+        self.account.save(claimedDictionary, to: path)
+    }
+
+    pub fun stakingEnabled(): Bool {
+        if let stakingEnabled = self.account.load<Bool>(from: /storage/stakingEnabled) {
+            self.account.save(stakingEnabled, to: /storage/stakingEnabled)
+            return stakingEnabled
+        } else {
+            return false
+        }
```

Comment:
Don't need to `load` (remove) and re-`save` just to read: Bool is a struct, you can use `copy`:

```suggestion
        return self.account.copy<Bool>(from: /storage/stakingEnabled) ?? false
```

Reply:
oh yeah I forgot this existed! Thanks for reminding me! :)